rust/src/transformations/sum/mod.rs

```diff
@@ -15,82 +15,52 @@ use crate::metrics::{AbsoluteDistance, InsertDeleteDistance, SymmetricDistance};
-/// * [DMNS06 Calibrating Noise to Sensitivity in Private Data Analysis](https://people.csail.mit.edu/asmith/PS/sensitivity-tcc-final.pdf)
-///
-/// # Arguments
-/// * `bounds` - Tuple of lower and upper bounds for data in the input domain.
-///
-/// # Generics
-/// * `MI` - Input Metric. One of `SymmetricDistance` or `InsertDeleteDistance`.
-/// * `T` - Atomic Input Type and Output Type.
-pub fn make_bounded_sum<MI, T>(
-    bounds: (T, T),
-) -> Fallible<Transformation<VectorDomain<AtomDomain<T>>, AtomDomain<T>, MI, AbsoluteDistance<T>>>
-where
-    MI: Metric,
-    T: MakeBoundedSum<MI>,
-    (VectorDomain<AtomDomain<T>>, MI): MetricSpace,
-{
-    T::make_bounded_sum(bounds)
-}
-
-#[bootstrap(
-    features("contrib"),
-    generics(MI(default = "SymmetricDistance"), T(example = "$get_first(bounds)")),
-    returns(c_type = "FfiResult<AnyTransformation *>")
-)]
-/// Make a Transformation that computes the sum of bounded data with known dataset size.
-///
-/// This uses a restricted-sensitivity proof that takes advantage of known dataset size for better utility.
-/// Use `make_clamp` to bound data and `make_resize` to establish dataset size.
+/// If dataset size is known, uses a restricted-sensitivity proof that takes advantage of known dataset size for better utility.
 ///
 /// # Citations
 /// * [CSVW22 Widespread Underestimation of Sensitivity...](https://arxiv.org/pdf/2207.10635.pdf)
 /// * [DMNS06 Calibrating Noise to Sensitivity in Private Data Analysis](https://people.csail.mit.edu/asmith/PS/sensitivity-tcc-final.pdf)
 ///
 /// # Arguments
-/// * `size` - Number of records in input data.
+/// * `input_domain` - Domain of the input data.
+/// * `input_metric` - One of `SymmetricDistance` or `InsertDeleteDistance`.
 /// * `bounds` - Tuple of lower and upper bounds for data in the input domain.
 ///
 /// # Generics
 /// * `MI` - Input Metric. One of `SymmetricDistance` or `InsertDeleteDistance`.
 /// * `T` - Atomic Input Type and Output Type.
-pub fn make_sized_bounded_sum<MI, T>(
-    size: usize,
-    bounds: (T, T),
+pub fn make_sum<MI, T>(
+    input_domain: VectorDomain<AtomDomain<T>>,
+    input_metric: MI,
 ) -> Fallible<Transformation<VectorDomain<AtomDomain<T>>, AtomDomain<T>, MI, AbsoluteDistance<T>>>
 where
     MI: Metric,
-    T: MakeSizedBoundedSum<MI>,
+    T: MakeSum<MI>,
     (VectorDomain<AtomDomain<T>>, MI): MetricSpace,
+    (AtomDomain<T>, AbsoluteDistance<T>): MetricSpace,
 {
-    T::make_sized_bounded_sum(size, bounds)
+    T::make_sum(input_domain, input_metric)
 }
 
-// make_(sized_)?bounded_sum
+// make_sum
 
 // implementations delegate to:
 // make_(sized_)?bounded_int_(checked|monotonic|ordered|split)_sum
 // make_(sized_)?bounded_float_(checked|ordered)_sum
 
 #[doc(hidden)]
-pub trait MakeBoundedSum<MI: Metric>: Sized + CheckAtom + Clone + TotalOrd
+pub trait MakeSum<MI: Metric>: Sized + CheckAtom + Clone + TotalOrd
```

Comment:
It's amazing that all the variations can be captured by a single trait (plus some clever macros)!